### DISCUSSION TITLE
Add more tests for the Repository and ViewModel layer

### DISCUSSION THREAD

**Discussion by sleepylee** [nimblehq/android-templates#319] [RFCs]
### What happened
I noticed that we did not add unit tests for the ViewModel and Repository layers. 

Oftentimes, these involve the async data processing, in the current context, is Flow/Coroutine/Suspend functions.

Not having these setups could be challenging for everyone to start a project with decent coverage.

https://github.com/nimblehq/android-templates/tree/develop/template/domain/src/test/java/co/nimblehq/template/domain/usecase
https://github.com/nimblehq/android-templates/blob/develop/template/app/src/test/java/co/nimblehq/template/test/ViewModelExt.kt

**Comment by luongvo** [comment]
@sleepylee Actually, we already had unit tests for:
- Repository: https://github.com/nimblehq/android-templates/blob/develop/template/data/src/test/java/co/nimblehq/template/data/repository/RepositoryTest.kt
- ResponseMapping: https://github.com/nimblehq/android-templates/blob/develop/template/data/src/test/java/co/nimblehq/template/data/extensions/ResponseMappingTest.kt
- UseCase: https://github.com/nimblehq/android-templates/blob/develop/template/domain/src/test/java/co/nimblehq/template/domain/usecase/UseCaseTest.kt

...only `ViewModel` layer is missing its test 👌

**Comment by luongvo** [comment]
@Tuubz Can we have more tickets for this RFC for closing? I think the best is to review all the template and see if there are any missing tests 🙏

**Reply by toby-thanathip** [reply]
Noted 🙏

**Reply by toby-thanathip** [reply]
I can confirm that the following test classes are missing:

**template-xml** & **sample-xml**:  `HomeViewModelTest.kt`

The rest has already been added, so yeah let's add it 🚀

**Reply by luongvo** [reply]
cc @ryan-conway

**Comment by chornerman** [comment]
Voted 🚀

**Comment by ryan-conway** [comment]
As we are [deprecating the XML template](https://github.com/nimblehq/android-templates/issues/503) and only XML tests are missing, I believe we can close this discussion 🙇🏻
